wandb/sdk/wandb_settings.py

```diff
@@ -674,6 +679,27 @@ def _path_convert(*args: str) -> str:
         """
         return os.path.expanduser(os.path.join(*args))
 
+    def _get_url_query_string(self) -> str:
+        if self.anonymous != "true":
+            return ""
+        return "?" + urlencode({"apiKey": self.api_key})
```

Comment:
it used to get the api_key from this function: 
https://github.com/wandb/client/blob/e113eb41eb973bde784d1585cddbe6e79b641396/wandb/sdk/lib/apikey.py#L222

make sure that it doesn't break any functionality

Reply:
Yea, this part is not taken care elsewhere (at least not in Settings), I think: `auth = requests.utils.get_netrc_auth(settings.base_url)`

Reply:
seems like it would be easy to add this logic